geowrangler/vector_zonal_stats.py

```diff
@@ -369,16 +369,19 @@ def create_bingtile_zonal_stats(
     _validate_aggs(fixed_aggs, data)
 
     # create aoi level quad_key for data (apply quadkey_to_tile)
+
     quadkey_zoom_tile = partial(quadkey_to_tile, zoom_level=aoi_zoom_level)
 
-    features = data.copy()
-    features[GEO_INDEX_NAME] = features[data_quadkey_column].apply(quadkey_zoom_tile)
+    aoi2 = aoi[[aoi_quadkey_column]].set_index(aoi_quadkey_column)
```

Comment:
Can you explain what is happening here? why do we need an aoi2?

Reply:
hi @jtmiclat maybe there's a better way to do it -- but the intention was to filter the data rows so that only the rows with the aoi_quadkey are left in the features (w/o adding a new column)

Reply:
Hi @jtmiclat I just cleaned up the statement to remove the temp variable since its only used once anyway. Still open to suggestions on how to filter the data so that only those rows which have their quadkeys within the the aoi quadkeys are left.